### PR TITLE
WIP: Drawing method for Lattice

### PR DESCRIPTION
**Pull request by VolodyaCO**
This PR fixes a floating point error that hid some edges when creating lattices, as suggested by @chrisrothUT in #611. The PR also creates a method to draw the lattice and its edges.

Some things to discuss:

1. Should I enforce that graphs are drawn only for 1D and 2D lattices?
2. I added a curvature by default to the edges, because when using pbcs some edges are drawn on top of other edges, making it impossible to differentiate them. Should I expose an argument to control the curvature?
3. What other arguments should I expose?

### CODE REVIEW

File: netket/graph/lattice.py [hunk 1]
```diff
@@ -19,6 +19,9 @@
 import itertools
 import networkx as _nx
 import warnings
+from typing import Tuple, Union
+import matplotlib
```
Comment (gcarleo) [inline netket/graph/lattice.py:23]: we don't want to depend on matplotlib, I think you should import this only if draw is called

File: netket/graph/lattice.py [hunk 2]
```diff
@@ -78,8 +81,9 @@ def create_points(basis_vectors, extent, atom_coords, pbc):
 
 def get_true_edges(basis_vectors, atoms, cellANDlabel_to_site, extent):
     atoms_positions = dicts_to_array(atoms, "r_coord")
+    tol = 1e-5
     naive_edges = get_edges(
-        atoms_positions, _np.linalg.norm(basis_vectors, axis=1).max()
+        atoms_positions, _np.linalg.norm(basis_vectors, axis=1).max() + tol
```
Comment (PhilipVinc) [inline netket/graph/lattice.py:86]: why you add the tolerance here?

File: netket/graph/lattice.py [hunk 3]
```diff
@@ -223,6 +227,38 @@ def atoms_coord(self):
         """
         return self._atoms_coord
 
+    def draw(
+        self, figsize: Tuple[Union[int, float]] = None, ax: matplotlib.axes.Axes = None
```
Comment (PhilipVinc) [inline netket/graph/lattice.py:231]:
```suggestion
        self, figsize: Optional[Tuple[Union[int, float]]] = None, ax: matplotlib.axes.Axes = None
```

File: netket/graph/lattice.py (hunk 1 above)
Comment (gcarleo) [reply netket/graph/lattice.py:23]: no wait, did we just merge this import matplotlib when importing lattice?!? @PhilipVinc

**Comment by gcarleo** [comment]
ah no it's ok, this was automatically closed but not merged

File: netket/graph/lattice.py (hunk 2 above)
Comment (VolodyaCO) [reply netket/graph/lattice.py:86]: Yeah, I'll have to check if the kdtree allows for this tolerance as an argument. If not I'll just add it inside get_edges as suggested.

**Comment by VolodyaCO** [comment]
I've got some time to finish this one today. I'm going to expose some plotting style options like colours, sizes and curvatures, and will move the matplotlib import inside the draw method. @gcarleo @PhilipVinc which branch should I merge this to? master?

**Comment by PhilipVinc** [comment]
Ah sorry @VolodyaCO , I merged the other PR and since you where targeting it's branch this was closed... I did not notice you where targeting the branch.
Just reopen it targeting the main branch.

**Pull request closed without merging**